open-api/rest-catalog-open-api.yaml

```diff
@@ -3103,6 +3103,22 @@ components:
         uuid:
           type: string
 
+    StorageCredential:
+      type: object
+      required:
+        - prefix
+        - config
+      properties:
+        prefix:
+          type: string
+          description: Indicates a storage location prefix where the credential is relevant. Clients should choose the most
+            specific prefix if several credentials of the same type are available.
+        config:
+          type: object
+          additionalProperties:
```

Comment:
I understand that there could be some common attributes that we could extract, but that would lead to a mixture of having to look up things in a config map vs other properties. 
Also what if storage providers have different semantics for how expiration is defined (expires-at-ms vs expires-in-ms vs ....). This would require additional calculations to convert from one expiration format to the other. Therefore I think it's better to keep everything in the config for now until we have a clear use case where it makes sense to extract this into a common attribute.

This isn't set in stone and we can always revisit this once we implement refreshing vended credentials for the different providers and see how helpful having a separate field for expiration would be. I'd like to unblock this proposal so that we can make progress on refreshing vended credentials